File: yunu/프로그래머스/시소 짝꿍/좋은 코드.js
function solution(weights) {
  weights.sort((a, b) => b - a);
  const cache = {};
  let cnt = 0;
  for (let n of weights) {
    let m;
    if (cache[n]) {
      cnt += cache[n];
    }
    if (((m = (n * 3) / 2), cache[m])) {
      cnt += cache[m];
    }
    if (((m = n * 2), cache[m])) {
      cnt += cache[m];
    }
    if (((m = (n * 4) / 3), cache[m])) {
      cnt += cache[m];
    }

    if (!cache[n]) cache[n] = 1;
    else cache[n]++;
  }

  return cnt;
}

console.log(solution([100, 180, 360, 100, 270]));

File: yunu/프로그래머스/호텔 대실/좋은 코드.py
def solution(book_time):
    room = 0
    for i in range(len(book_time)):
        it = list(map(int, book_time[i][0].split(':')))
        et = list(map(int, book_time[i][1].split(':')))
        book_time[i] = [it[0]*60 + it[1], et[0]*60 + et[1]]
    book_time.sort()
    while book_time:
        room += 1
        std = book_time.pop(0)
        for book in book_time[:]:
            if book[0] >= std[1]+10: 
                book_time.remove(book)
                std = book
    return room